### ISSUE TITLE
Provide an ability to annotate locale for accessibility

### ISSUE THREAD

**Issue by vick08** [flutter/flutter#53972]
## Use case
Developers should be able to specify the accessibility locale for UI elements, including text, in their apps. This information could then be used by assistive technology software to either switch TTS languages  on the fly or perhaps load different Braille table based on the specified   locale.

If a accessibility layer is not aware that the element is in a different locale, for example, the screen reader may pronounce it using system default locale, which may be different from the one the app is using.

## Proposal

Both iOS and Android provide a way to annotate locale via [accessibilityLanguage](https://developer.apple.com/documentation/objectivec/nsobject/1615192-accessibilitylanguage) and [LocaleSpan](https://developer.android.com/reference/android/text/style/LocaleSpan) respectively.
It would be great for Flutter to provide a similar semantic affordance for developers.

**Comment by hkwinslow** [comment]
Any timeline for this?

**Comment by chunhtai** [comment]
You can now do
```dart
Text.rich(TextSpan(text: <some localized string>, locale: <Locale>))
```
or
```dart
Semantics(
  attributedLabel: AttributedString( <some localized string>, attributes: [LocaleStringAttribute(...)])
)
```
